alert/tests.py

```diff
@@ -532,21 +534,190 @@ def test_no_permission(self):
+        self.assertEqual(response.data['email'], None)
+        self.assertEqual(response.data['phone'], '+13131234567')
+        self.assertEqual(response.data['user']['username'], 'jacob')
+        self.assertEqual(response.data['user']['first_name'], '')
+        self.assertEqual(response.data['user']['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['email'], None)
+        self.assertEqual(response.data['phone'], '+13131234567')
+        self.assertEqual(response.data['user']['username'], 'jacob')
+        self.assertEqual(response.data['user']['first_name'], '')
+        self.assertEqual(response.data['user']['last_name'], '')
+
+    def test_null_phone(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'email': 'example@email.com',
+                                               'phone': None}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['email'], 'example@email.com')
+        self.assertEqual(response.data['phone'], None)
+        self.assertEqual(response.data['user']['username'], 'jacob')
+        self.assertEqual(response.data['user']['first_name'], '')
+        self.assertEqual(response.data['user']['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['email'], 'example@email.com')
+        self.assertEqual(response.data['phone'], None)
+        self.assertEqual(response.data['user']['username'], 'jacob')
+        self.assertEqual(response.data['user']['first_name'], '')
+        self.assertEqual(response.data['user']['last_name'], '')
+
+    def test_both_null(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'email': None,
+                                               'phone': None}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['email'], None)
+        self.assertEqual(response.data['phone'], None)
+        self.assertEqual(response.data['user']['username'], 'jacob')
+        self.assertEqual(response.data['user']['first_name'], '')
+        self.assertEqual(response.data['user']['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['email'], None)
+        self.assertEqual(response.data['phone'], None)
+        self.assertEqual(response.data['user']['username'], 'jacob')
+        self.assertEqual(response.data['user']['first_name'], '')
+        self.assertEqual(response.data['user']['last_name'], '')
```

Comment:
Should also have tests to make sure that users cannot access the data of other users. Some examples I can think of off the top of my head:
- Can't read/update user contact info
- Can't see another user's registrations

Views should be returning `403 Forbidden` when data is inaccessible to them. You can return a django `HttpForbidden()` object

Reply:
Does test_multiple_users_independent cover what you're getting at?  I'm not sure logistically how a user would access another user's data so I'm not really sure when to return a 403.  See above comment.